llvm/lib/Target/RISCV/RISCVFoldMasks.cpp

```diff
@@ -133,6 +438,48 @@ bool RISCVFoldMasks::convertVMergeToVMv(MachineInstr &MI, MachineInstr *V0Def) {
   return true;
 }
 
+bool RISCVFoldMasks::convertToUnmasked(MachineInstr &MI,
```

Comment:
This transform is much simpler and easy to review.  What happens if we move only this one to start with?  Does it cause real regressions?

Reply:
Unfortunately convertToUnmasked is the one with the very large test diff. I think it also runs before `doPeepholeMergeVVMFold` so it would affect the output of that too

Reply:
You can see the test diff here: https://github.com/lukel97/llvm-project/commit/01854599f2cfd84899c37f4791d6c4a0784017e1

Reply:
I didn't understand this, so I took my own shot at it (stealing your code as a starting point).  https://github.com/llvm/llvm-project/pull/72441 shows the result, and has a fixme which explains the coupling.  We could either land the half split state (my change), or leave it rolled in here.